addons/payment_stripe/models/payment.py

```diff
@@ -149,14 +161,97 @@ def _get_feature_support(self):
+        record in Odoo.
+        """
+        SUPPORTED_WEBHOOKS = ['checkout.session.completed']
+        wh_type = data.get('type')
+        if wh_type not in SUPPORTED_WEBHOOKS:
+            _logger.info('unsupported webhook type %s, ignored', wh_type)
+            return False
+        else:
+            _logger.info('handling %s webhook event from stripe', wh_type)
+
+        signature = request.httprequest.headers.get('Stripe-Signature')
+        body = request.httprequest.data
+        self._verify_stripe_signature(signature, body)
+
+        stripe_object = data.get('data', {}).get('object')
+        if not stripe_object:
+            raise ValidationError('Stripe Webhook data does not conform to the expected API.')
+        if wh_type == 'checkout.session.completed':
+            return self._handle_checkout_webhook(stripe_object)
+        return False
+
+    def _verify_stripe_signature(self, signature: str, body: bytes):
+        """
+        :param signature: t=xxx,v0=xxx,v1=xxx string provided in request header
+        :param body: bytes provided as the request body
+        :return: true if and only if signature matches hash of payload calculated with secret
+        :raises ValidationError: if signature doesn't match
+        """
+        sign_data = {k: v for (k, v) in [s.split('=') for s in signature.split(',')]}
+        event_timestamp = int(sign_data['t'])
+        if datetime.utcnow().timestamp() - event_timestamp > STRIPE_SIGNATURE_AGE_TOLERANCE:
+            _logger.error('stripe event is too old, event is discarded')
+            raise ValidationError('event timestamp older than tolerance')
+
+        signed_payload = "%s.%s" % (event_timestamp, body.decode('utf-8'))
+
+        actual_signature = sign_data['v1']
+        expected_signature = hmac.new(self.stripe_webhook_secret.encode('utf-8'),
+                                      signed_payload.encode('utf-8'),
+                                      sha256).hexdigest()
+
+        if not consteq(expected_signature, actual_signature):
+            _logger.error(
+                'incorrect webhook signature from Stripe, check if the webhook signature '
+                'in Odoo matches to one in the Stripe dashboard')
+            raise ValidationError('incorrect webhook signature')
+
+        return True
+
+    def _handle_checkout_webhook(self, checkout_object: dir):
+        """
+        Process a checkout.session.completed Stripe web hook event,
+        mark related payment successful
+
+        :param checkout_object: provided in the request body
+        :return: True if and only if handling went well, False otherwise
+        :raises ValidationError: if input isn't usable
+        """
+        tx_reference = checkout_object.get('client_reference_id')
+        data = {'reference': tx_reference}
+        try:
+            odoo_tx = self.env['payment.transaction']._stripe_form_get_tx_from_data(data)
```

Comment:
you probably need to verify the signature *after* this, since in the webhook call, we don't know yet which acquirer will be handling the signature check (you might have more than one stripe account configured, in which case you must first find which tx the event is for, then use the tx's acquirer to do the signature check)

Reply:
Ah my bad, I didn't think of that, I thought I would check asap to avoid any unnecessary work.

Will do.

Reply:
I've modified `_verify_stripe_signature` so that it no longer requires parameters. Not a big fan as it increases its responsibility but apart from complicating the tests, it's not bad atm. An alternative would have been passing a nested method from `_handle_stripe_webhook` to `_handle_checkout_webhook`.

**`_handle_checkout_webhook` verifies signature at after finding the correct acquirer instance.**